src/api/telemetry/TelemetryAPI.js

```diff
@@ -419,6 +441,26 @@ export default class TelemetryAPI {
       subscriber.callbacks.push(callback);
     }
 
+    function subscriptionCallbackForArray(value) {
+      if (!Array.isArray(value)) {
+        value = [value];
+      }
+
+      subscriber.callbacks.forEach(function (cb) {
+        cb(value);
+      });
+    }
+
+    function subscriptionCallbackForSingleValue(value) {
+      if (Array.isArray(value)) {
+        value = value[value.length - 1];
```

Comment:
will this ever be a zero length array?

Reply:
`value` would evaluate to `undefined`.  Perhaps we should just have a general policy of never invoking a subscription callback without a telemetry datum. I'll add some code to enforce that.

Reply:
Done.